From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Mon, 23 May 2022 15:52:19 +0100
Subject: linux limits

Should not #include <linux/limits.h> on non-Linux platforms.
In fact, as shown below, #include <limits.h> on Linux includes linux/limits.h, so just do that.

$ echo '#include <limits.h>' > /tmp/l.c
$ g++ -E /tmp/l.c | egrep linux/limits
# 1 "/usr/include/linux/limits.h" 1 3 4
---
 src/core/util_string.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/core/util_string.cpp b/src/core/util_string.cpp
index aa4c2d1..f175ed6 100644
--- a/src/core/util_string.cpp
+++ b/src/core/util_string.cpp
@@ -37,9 +37,7 @@
 #include <Windows.h>
 #endif // __PLATFORM_WIN32__
 
-#ifdef __PLATFORM_LINUX__
-#include <linux/limits.h>
-#endif // __PLATFORM_LINUX__
+#include <limits.h>
 
 #include <stdio.h>
 using namespace std;
